controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -494,10 +496,35 @@ func (r *BareMetalHostReconciler) actionDeleting(prov provisioner.Provisioner, i
 		return deleteComplete{}
 	}
 
+	// Only try to power off `maxPowerOffRetryCount` times before giving up, and deleting the host anyway
+	if info.host.Status.ErrorType != metal3v1alpha1.PowerManagementError || (info.host.Status.ErrorType == metal3v1alpha1.PowerManagementError && info.host.Status.ErrorCount < maxPowerOffRetryCount) {
+		info.log.Info("host ready to be powered off")
+		provResult, err := prov.PowerOff(
+			metal3v1alpha1.RebootModeHard,
+			info.host.Status.ErrorType == metal3v1alpha1.PowerManagementError)
+
+		if err != nil {
+			return actionError{errors.Wrap(err, "failed to power off")}
+		}
+
+		if provResult.ErrorMessage != "" {
+			return recordActionFailure(info, metal3v1alpha1.PowerManagementError, provResult.ErrorMessage)
+		}
+
+		if provResult.Dirty {
+			return actionContinue{provResult.RequeueAfter}
+		}
+	}
+
 	provResult, err := prov.Delete()
 	if err != nil {
 		return actionError{errors.Wrap(err, "failed to delete")}
 	}
+
+	if provResult.ErrorMessage != "" {
+		return recordActionFailure(info, metal3v1alpha1.DeleteError, provResult.ErrorMessage)
+	}
```

Comment:
I'm not against adding this (and the DeleteError constant in general) for completeness (a hypothetical non-ironic provisioner could potentially fail here, even though the ironic one never does). But if we do I'd prefer we did it in a separate PR (or at least a separate commit) so it is clear to future-us that this is independent of the power-off-on-deletion thing.